pyogrio/raw.py

```diff
@@ -84,19 +85,36 @@ def read(
             "geometry": "<geometry type>"
         }
     """
+    from_buffer = False
+    if isinstance(path_or_buffer, bytes):
```

Comment:
Here I am checking for bytes vs strings to determine whether it's a path or in-memory bytes. I don't know if that is robust enough? Or do we want a separate `read_buffer` or so? 
Alternatively (or in addition), we could also support "file-like" objects (objects that have a `read()` method that will return the bytes). That's eg what fiona does in their `open()` method.

Reply:
I think file-like objects that have a `read()` method would provide a more general solution?  For example, the `is_zipped` below check is limited to a single zip format, whereas the file-like pattern would let user construct and pass in a `ZipFile` or `GzipFile` class instance.  I'm not familiar with fsspec, but it seems like the file-like pattern would support that as well.